doc/spec_dataValidityPropagation.dox

```diff
@@ -85,6 +87,49 @@ See @ref spec_execptionHandling.
 
+4. Circular dependencies
+------------------------
+
+If modules have circular dependencies, the algorithm described in \ref dataValidity_1 "section 1" leads to a self-excited loop: Once the DataValidity::invalid flag has made a full circle, there is always at least on input with invalid data in each module and you can't get rid of it any more. To break this circle, the following additional behaviour is implemented:
+
+### 4.1 General behaviour
+
+- \anchor dataValidity_4_1_1 4.1.1 Inputs which are part of a circular dependency are marked as _circular input_.[\ref dataValidity_test_TestCircularInputDetection "T"]
+    - \anchor dataValidity_4_1_1_1 4.1.1.1 Inputs which are coming from other applicatiation modules
+ which are not part of the circle, from the control system module or from device modules are considered _external inputs_.[\ref dataValidity_test_TestCircularInputDetection "T" (only CS module)]
+- \anchor dataValidity_4_1_2 4.1.2 All modules which have a circular dependency form a _circular network_.[\ref dataValidity_test_TestCircularInputDetection "T"]
+     - \anchor dataValidity_4_1_2_1 4.1.2.1 Also entangled circles of different variables which intersect in some of the modules are part of the same circular network.[\ref dataValidity_test_TestCircularInputDetection2 "T"]
+     - \anchor dataValidity_4_1_2_2 4.1.2.2 There can be multiple disconnected circular networks in an application.[\ref dataValidity_test_TestCircularInputDetection2 "T"]
+- 4.1.3 Circular inputs and circular networks are identified at application start after the variable networks are established.
+- \anchor dataValidity_4_1_4 4.1.4 As long as at least one _external input_ of any module in the _circular network_ is invalid, the invalidity flag is propagated as described in
+ \ref dataValidity_1 "1." [\ref dataValidity_test_OneInvalidVariable "T"]
+- \anchor dataValidity_4_1_5 4.1.5 Once all _external inputs_ of one _circular network_ are back to DataValidity::ok, all _circular inputs_ of the _circular network_ ignore the
+invalid flag and also switch to DataValidity::ok. This breaks the circle. [\ref dataValidity_test_OneInvalidVariable "T"] [\ref dataValidity_test_TwoFaultyInOneModule "T"]
+- \anchor dataValidity_4_1_6 4.1.6 If all inputs of a module have DataValidity::ok, the module's output validity is also DataValidity::ok, even if other modules in
+the _circular network_ have _external inputs_ which are invalid.
+- \anchor dataValidity_4_1_7 4.1.7 The ControlSystemModule and DeviceModules are never part of a circular dependency.[\ref dataValidity_test_TestCircularInputDetection2 "T" (only CS module)]
+
+### 4.2 Side effects and race conditions
+
+- \anchor dataValidity_4_2_1 4.2.1 If a process variable has a fluctuating error state (e.g. alternating validity `ok` and `faulty`) and is the only external input, the faulty flag
```

Comment:
I don't see the connection to fluctuating error states here. When all external inputs are back to valid, there will be always still data in the circle which is based off invalid data, but this fact is ignored. This happens whenever the state goes back to valid, whether it fluctuates or not does not matter.